standard attr page - [WEB-3769] (#19087)
DataDog/documentation: content/en/standard-attributes/_index.md

---
title: Default Standard Attributes
disable_sidebar: true
further_reading:
    - link: '/data_security/'
      tag: 'Documentation'
      text: 'Ensuring the security of the data sent to Datadog'
filter_all: All
content: "The following table lists the attributes automatically applied to data sent to Datadog by the Agent by each of the RUM, Logs, and APM products, as applicable to the data domain. Optionally, filter the list by product or search by keyword or description text to find the attributes you're interested in."

> attributes:

Reviewer: Is there a way to bump up the padding on the page so the metric description table isn't cut off slightly?

![image](https://github.com/DataDog/documentation/assets/76412946/7cf96d70-bf42-445f-b2da-227a757246bc)

Author: you should be able to stroll the table to the left or right, but I'll tinker with the table a bit

> attributes:

Reviewer 2: is this data aggregated from somewhere that we can automate?    this is a massive list, just worried it will be difficult to maintain over time.

Reviewer 3: @bgdeutsch It's not, but I would love to. Not sure how to make that happen. Just getting this page made was job 1. (We're already maintaining these manually in each of the product docs pages -- I don't think they change all that often)

  - name: host
    description: The name of the originating host as defined in metrics. Datadog automatically retrieves corresponding host tags from the matching host in Datadog and applies them to your telemetry. The Agent sets this value automatically.
    product_source: 
      - icon-log
      - icon-apm
    type: string
    domain: Reserved
  - name: device
    description: The type of originating device.
    product_source:
      - icon-rum
      - android
      - ios
      - browser
      - roku
    type: string
    domain: Reserved
  - name: source
    description: This corresponds to the integration name, the technology from which the data originated. When it matches an integration name, Datadog automatically installs the corresponding parsers and facets. For example, `nginx`, `postgresql`, and so on.
    product_source: 
      - icon-log
      - icon-apm
    type: string
    domain: Reserved
  - name: status

>     description: This corresponds to the level/severity of the data. For logs, it is used to define [log patterns](/logs/explorer/patterns/) and has a dedicated layout in the Datadog Log UI.

Reviewer: ```suggestion
    description: This corresponds to the level or severity of the data. For logs, it is used to define [log patterns](/logs/explorer/patterns/) and has a dedicated layout in the Log Management UI.
```

    product_source: 
      - icon-log
      - icon-apm
    type: string
    domain: Reserved
  - name: service

>     description: The [unified service name](/getting_started/tagging/unified_service_tagging/) for the application or service that is generating the data, used to correlate user sessions. It is used to switch from APM to other products, so make sure you define the same value when you use both products. In RUM browser SDK, a service denotes a set of pages built by a team that offers a specific functionality in your browser application. You can assign web pages to a service with [manual view tracking](/real_user_monitoring/browser/modifying_data_and_context/?tab=npm#override-default-rum-view-names).

Reviewer: ```suggestion
    description: The [unified service name](/getting_started/tagging/unified_service_tagging/) for the application or service that is generating the data, used to correlate user sessions. It is used to switch from APM to other products, so make sure you define the same value when you use both products. In the RUM Browser SDK, a service denotes a set of pages built by a team that offers a specific functionality in your browser application. You can assign web pages to a service with [manual view tracking](/real_user_monitoring/browser/modifying_data_and_context/?tab=npm#override-default-rum-view-names).
```

    description: "A database statement for the given database type. For example, for mySQL: `'SELECT * FROM wuser_table';` and for Redis: `'SET mykey 'WuValue''`."
    product_source: 
      - icon-log
    type: string
    domain: Database
  - name: db.operation
    description: The operation that was performed ("query", "update", "delete", and so on).
    product_source: 
      - icon-log
    type: string
    domain: Database
  - name: db.user
    description: User that performs the operation.
    product_source: 
      - icon-log
    type: string
    domain: Database
  - name: duration
    description: "A duration of any kind in **nanoseconds**: HTTP response time, database query time, latency, and so on. [Remap](/logs/log_configuration/processors/#remapper) any durations within logs to this attribute because Datadog displays and uses it as a default measure for trace search."
    product_source: 
      - icon-log
    type: number
    domain: Performance
  - name: usr.id
    description: The user identifier.
    product_source: 
      - icon-log 
      - icon-rum
      - android
      - ios
      - browser
      - roku
    type: string
    domain: User
  - name: usr.name

>     description: The user friendly name.

Reviewer: ```suggestion
    description: The user-friendly name.
```

  
  - name: session.initial_view.name
    description: Name of the initial view of the session.
    product_source:
      - icon-rum
      - android
      - ios
      - roku
    type: string
    domain: Session (Android events, iOS events, Roku events)
  
  - name: session.last_view.url
    description: URL of the last view of the session.
    product_source:
      - icon-rum
      - android
      - ios
      - roku
    type: string
    domain: Session (Android events, iOS events, Roku events)
  
  - name: session.last_view.name
    description: Name of the last view of the session.
    product_source:
      - icon-rum
      - android
      - ios
      - roku
    type: string
    domain: Session (Android events, iOS events, Roku events)
  
  - name: session.ip
    description: IP address of the session extracted from the TCP connection of the intake. If you want to stop collecting   this attribute, change the setting in your [application details](/data_security/real_user_monitoring/#ip-address).
    product_source:
      - icon-rum
      - android
      - ios
      - roku
    type: string
    domain: Session (Android events, iOS events, Roku events)
  
  - name: session.useragent

>     description: System user agent info to interpret device info.

Reviewer: ```suggestion
    description: System `User-Agent` info to interpret device info.
```

    product_source:
      - icon-rum
      - browser
    type: string
    domain: Operating system (Browser)
  - name: os.version_major
    description: The OS version major as reported by the device (User-Agent HTTP header).
    product_source:
      - icon-rum
      - browser
    type: string
    domain: Operating system (Browser)

  - name: session.id
    description: Randomly generated ID for each session.
    product_source:
      - icon-rum
      - browser
    type: string
    domain: Session (Browser events)
  
  - name: session.ip
    description: Client IP address. If you want to stop collecting this attribute, change the setting in your [application details](/data_security/real_user_monitoring/#ip-address).
    product_source:
      - icon-rum
      - browser
    type: string
    domain: Session (Browser events)
  
  - name: session.is_active
    description: Indicates if the session is currently active. The session ends after 4 hours of activity or 15 minutes of inactivity.
    product_source:
      - icon-rum
      - browser
    type: boolean
    domain: Session (Browser events)
  
  - name: session.type

>     description: The type of session, `user` or `synthetics`. Sessions from [Synthetic Monitoring Browser Tests](/synthetics/browser_tests/) are excluded from billing.

Reviewer: ```suggestion
    description: The type of session, `user` or `synthetics`. Sessions from [Synthetic Browser Tests](/synthetics/browser_tests/) are excluded from billing.
```

    description: The parameter in the URL tracking the source of traffic.
    product_source:
      - icon-rum
      - browser
    type: string
    domain: UTM (Browser events)
  
  - name: view.url_query.utm_medium
    description: The parameter in the URL tracking the channel where the traffic is coming from.
    product_source:
      - icon-rum
      - browser
    type: string
    domain: UTM (Browser events)
  
  - name: view.url_query.utm_campaign
    description: The parameter in the URL identifying the specific marketing campaign tied to that view.
    product_source:
      - icon-rum
      - browser
    type: string
    domain: UTM (Browser events)
  
  - name: view.url_query.utm_content
    description: The parameter in the URL identifying the specific element a user clicked within a marketing campaign.
    product_source:
      - icon-rum
      - browser
    type: string
    domain: UTM (Browser events)
  
  - name: view.url_query.utm_term
    description: The parameter in the URL tracking the keyword a user searched to trigger a given campaign.
    product_source:
      - icon-rum
      - browser
    type: string
    domain: UTM (Browser events)
  

  - name: language

>     description: The client SDK language used to generate the span. It can be one of cpp, dotnet, go, jvm, javascript, php, python, ruby.

Reviewer: ```suggestion
    description: The client SDK language used to generate the span. It can be one of `cpp`, `dotnet`, `go`, `jvm`, `javascript`, `php`, `python`, `ruby`.
```

    product_source:
      - icon-apm
    type: string
    domain: APM core
  - name: env

>     description: The value of `DD_ENV` environment variable or user defined `env` for the running process.

Reviewer: ```suggestion
    description: The value of the `DD_ENV` environment variable or user-defined `env` for the running process.
```

    product_source:
      - icon-apm
    type: string
    domain: APM core (Reserved)
  - name: version

>     description: The value of `DD_VERSION` environment variable or user defined `version` for the running process.

Reviewer: ```suggestion
    description: The value of the `DD_VERSION` environment variable or user-defined `version` for the running process.
```

    product_source:
      - icon-apm
    type: string
    domain: APM core (Reserved)
  - name: span.kind

>     description: The string representing the type of work unit handled by the span. It can be   one of server, client, producer, consumer or internal. More information in the   OpenTelemetry SpanKind documentation.

Reviewer: ```suggestion
    description: The string representing the type of work unit handled by the span. It can be one of server, client, producer, consumer, or internal. For more information, see the [OpenTelemetry SpanKind documentation](https://opentelemetry.io/docs/specs/otel/trace/api/#spankind).
```

    product_source:
      - icon-apm
    type: string
    domain: APM core
  - name: component

>     description: The name of the library/integration that created the span.

Reviewer: ```suggestion
    description: The name of the library or integration that created the span.
```

  - name: network.client.transport
    description: The transport protocol used to make the inbound connection.
    product_source:
      - icon-apm
    type: string
    domain: Network communications
  - name: network.destination.transport
    description: The transport protocol used to make the outbound connection.
    product_source:
      - icon-apm
    type: string
    domain: Network communications
  - name: http.status_code
    description: The HTTP response status code.
    product_source:
      - icon-apm
    type: string
    domain: HTTP requests
  - name: http.url
    description: The URL of the HTTP request, including the obfuscated query string. For more information on obfuscation, see Configure Data Security.
    product_source:
      - icon-apm
    type: string
    domain: HTTP requests
  - name: http.version
    description: The version of HTTP used for the request.
    product_source:
      - icon-apm
    type: string
    domain: HTTP requests
  - name: http.method
    description: The port of the client that initiated the connection.
    product_source:
      - icon-apm
    type: string
    domain: HTTP requests
  - name: http.route

>     description: The matched route (path template). For example, `/users/:userID`

Reviewer: ```suggestion
    description: The matched route (path template). For example, `/users/:userID`.
```

    product_source:
      - icon-apm
    type: string
    domain: HTTP requests
  - name: http.client_ip

>     description: The IP address of the original client behind all proxies, if   known. Discovered from headers such as X-Forwarded-For.

Reviewer: ```suggestion
    description: The IP address of the original client behind all proxies, if known. Discovered from headers such as `X-Forwarded-For`.
```

    product_source:
      - icon-apm
    type: string
    domain: HTTP requests
  - name: http.useragent

>     description: The user agent header received with the request.

Reviewer: ```suggestion
    description: The `User-Agent` header received with the request.
```

    product_source:
      - icon-apm
    type: string
    domain: HTTP requests
  - name: http.request.content_length
    description: The size of the request payload body in bytes.
    product_source:
      - icon-apm
    type: number
    domain: HTTP requests
  - name: http.response.content_length
    description: The size of the response payload body in bytes.
    product_source:
      - icon-apm
    type: number
    domain: HTTP requests
  - name: http.request.content_length_uncompressed

>     description: The size of the uncompressed request payload body after   transport decoding.

Reviewer: ```suggestion
    description: The size of the uncompressed request payload body after transport decoding.
```

    product_source:
      - icon-apm
    type: number
    domain: HTTP requests
  - name: http.response.content_length_uncompressed

>     description: The size of the uncompressed response payload body after   transport decoding.

Reviewer: ```suggestion
    description: The size of the uncompressed response payload body after transport decoding.
```

    product_source:
      - icon-apm
    type: number
    domain: HTTP requests
  - name: http.request.headers.*

>     description: The request HTTP headers. None are collected by default, but can   be optionally configured with DD_TRACE_HEADER_TAGS.To learn more about how to collect   headers, see the corresponding Library configuration.

Reviewer: I am assuming we can drop the "see the corresponding Library configuration" doc link, feel free to disregard.

```suggestion
    description: The request HTTP headers. None are collected by default, but can be optionally configured with `DD_TRACE_HEADER_TAGS`. 
```

    product_source:
      - icon-apm
    type: string
    domain: HTTP requests
  - name: db.system

>     description: Identifier for the database management system (DBMS product being used).

Reviewer: ```suggestion
    description: Identifier for the database management system (DBMS) product being used.
```

    product_source:
      - icon-apm
    type: string
    domain: Database spans
  
  - name: db.connection_string
    description: The connection string used to connect to the database.
    product_source:
      - icon-apm
    type: string
    domain: Database spans
    
  - name: db.user
    description: The username that accessed the database.
    product_source:
      - icon-apm
    type: string
    domain: Database spans
    
  - name: db.instance
    description: The name of the database being connected to.
    product_source:
      - icon-apm
    type: string
    domain: Database spans
    
  - name: db.statement
    description: The database statement being executed.
    product_source:
      - icon-apm
    type: string
    domain: Database spans
    
  - name: db.operation

>     description: The name of the operation being executed. For example, `SELECT`, `findAndModify`, `HMSET`

Reviewer: ```suggestion
    description: The name of the operation being executed. For example, `SELECT`, `findAndModify`, `HMSET`.
```

---
title: Default Standard Attributes
disable_sidebar: true
further_reading:
    - link: '/data_security/'
      tag: 'Documentation'
      text: 'Ensuring the security of the data sent to Datadog'
filter_all: All

> content: "Welcome to Datadog's Data Dictionary page! Here, you'll find a comprehensive and organized resource that unlocks the power of data understanding within the Datadog platform.<br></br>With our Data Dictionary, you'll gain insights into data sources, understand the meaning and context of each data point, and establish consistent naming conventions and definitions across your organization. This shared understanding eliminates ambiguity, fosters collaboration, and accelerates data-driven decision-making."

Reviewer 3: ```suggestion
content: "The following table lists the attributes automatically applied to data sent to Datadog by the Agent by each of the RUM, Logs, and APM products, as applicable to the data domain. Optionally, filter the list by product or search by keyword or description text to find the attributes you're interested in."
```
